saleor/order/base_calculations.py

```diff
@@ -21,29 +22,67 @@ def base_order_shipping(order: "Order") -> Money:
     return channel_listing.price
 
 
+def _base_order_subtotal(order: "Order", lines: Iterable["OrderLine"]) -> Money:
+    currency = order.currency
+    subtotal = zero_money(currency)
+    for line in lines:
+        quantity = line.quantity
+        price_with_discounts = line.base_unit_price * quantity
+        subtotal += price_with_discounts
+    return subtotal
+
+
 def base_order_total(order: "Order", lines: Iterable["OrderLine"]) -> Money:
     currency = order.currency
-    total = base_order_total_without_order_discount(order, lines)
+    subtotal = _base_order_subtotal(order, lines)
+    shipping_price = base_order_shipping(order)
     order_discounts = order.discounts.all()
     for order_discount in order_discounts:
-        total = apply_discount_to_value(
-            value=order_discount.value,
-            value_type=order_discount.value_type,
-            currency=currency,
-            price_to_discount=total,
-        )
-    return max(total, zero_money(currency))
+        if order_discount.type == OrderDiscountType.VOUCHER:
+            subtotal = apply_discount_to_value(
+                value=order_discount.value,
+                value_type=order_discount.value_type,
+                currency=currency,
+                price_to_discount=subtotal,
+            )
+        else:
+            if order_discount.value_type == DiscountValueType.PERCENTAGE:
```

Comment:
this should be `elif order_discount.value_type == DiscountValueType.PERCENTAGE:`

Reply:
Done in #10591